src/reactors/PriorityOrderReactor.sol

```diff
@@ -56,17 +67,41 @@ contract PriorityOrderReactor is BaseReactor {
         );
     }
 
+    /// @notice update the priority order with the cosigner's data
+    /// @dev only called if the current block is before the auctionStartBlock signed by the user
+    /// @param orderHash the hash of the order
+    /// @param order the order to update
+    function _updateWithCosignerData(bytes32 orderHash, PriorityOrder memory order) internal pure {
+        /// return quickly if cosignerData is not set
+        if (order.cosignerData.auctionTargetBlock == 0) return;
+
+        if (order.cosignerData.auctionTargetBlock < order.auctionStartBlock) {
+            order.auctionStartBlock = order.cosignerData.auctionTargetBlock;
```

Comment:
Looking at this more, I wonder if we should just combine all of this with `validate` and if we actually even need to update the `auctionStartBlock` field in the order? It's only ever used for validation purposes afaict. 

```
...deadline check and stuff

uint256 auctionStartBlock = user's
if (not past the user's start block, and a cosigner is specified) check sig + update the local
check local against current, throw `OrderNotFillable`
```

Reply:
that is true! we don't really need to actually override the field ... on second look wait this might break P2 verification bc the order hash will change? and it's not caught by forge

Reply:
looks like you hash the order before overriding the field but yeah still cleaner to not update it imo

Reply:
ah yea you're right